internal/rbd/rbd_util.go

```diff
@@ -468,11 +471,10 @@ func (ri *rbdImage) isInUse() (bool, error) {
 	return len(watchers) > defaultWatchers, nil
 }
 
-// checkImageFeatures check presence of imageFeatures parameter. It returns true when
-// there imageFeatures is missing or empty, skips missing parameter for non-static volumes
-// for backward compatibility.
-func checkImageFeatures(imageFeatures string, ok, static bool) bool {
-	return static && (!ok || imageFeatures == "")
+// checkValidImageFeatures check presence of imageFeatures parameter. It returns false when
+// there imageFeatures is present and empty.
+func checkValidImageFeatures(imageFeatures string, ok bool) bool {
+	return !(ok && imageFeatures == "")
```

Comment:
Technically, on the RBD side, an empty image features string is valid and means "create an image with no features enabled".  However disallowing it in the CSI layer is probably fine -- no one should be doing that these days.

Reply:
Yes, correct. this is extra validation to make sure the key `imageFeature` set in the storageclass is not empty as its optional user can choose to not set this key and value or set both.